indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1271,6 +1277,113 @@ public void remove(final Task task)
     }
   }
 
+  /**
+   * Acquire a read lock to perform the segment transactional append action for a given datasource.
+   * Also verifies that all the locks are of the type APPEND for the task.
+   * @param task task to perform the append action
+   * @param lockAcquireTimeoutMillis milliseconds to wait for lock acquisition
+   */
+  public void acquireTransactionalAppendLock(Task task, long lockAcquireTimeoutMillis)
+  {
+    for (TaskLockPosse lockPosse : findLockPossesForTask(task)) {
+      if (lockPosse.taskLock.getType() != TaskLockType.APPEND) {
+        throw new ISE(
+            "All the locks must be of type APPEND for segmentTransactionalAppend. Found lock of type[%s] for task[%s].",
+            lockPosse.taskLock.getType(), task.getId()
+        );
+      }
+    }
```

Comment:
This validation should be in a different method.

Reply:
Done